### ISSUE TITLE
How to Mint After Deploying Contract?

### ISSUE THREAD

**Issue by justinhchae** [PatrickAlphaC/all-on-chain-generated-nft#18]
Great repo! I followed along with your YouTube video and everything seems to work but I am not able to generate a token that I can look up on Opensea. I am not sure if I am missing a step or if the process is just taking some time to finish.

The issue is that when I run the scripts to generate a contract from an SVG, a contract is generated but a token is never minted. For example, when I go to Opensea and look for the contract (as you do early in the video at 12:15), nothing shows up for my test project when I perform the equivalent steps. However, the Etherscan seems to verify that the transaction went through.  Link to my last contract's Etherscan: [https://rinkeby.etherscan.io/address/0xdd73138E76A04f5392c776147F0725A1b7E5867D](https://rinkeby.etherscan.io/address/0xdd73138E76A04f5392c776147F0725A1b7E5867D) 

I followed along with the quick start instructions and my only deviation was to use a different SVG file (simple cartoon of a cat) to see if something with a bit more complexity would work. I should note that I got the same results with the circle SVG example you provide. I thought that the call to _safeMint() in the creat function should perform this task but the token counter is at 0 when I look up the metadata online - not sure how to debug this or where to start. 

I am still learning so please excuse me if I am not asking for the correct thing but I guess my question is this: On a testnet such as Rinkeby and after I successfully create the contract and verify it on Etherscan, how do I actually mint a token? 

Some CLI outputs and screencaps to illustrate what has worked and what has not:

```
# output from CLI when running npx hardhat deploy --network rinkeby --taks svg
You have deployed an NFT contract to 0xdd73138E76A04f5392c776147F0725A1b7E5867D
```
![etherscan](https://user-images.githubusercontent.com/68864597/151644143-6232541f-c941-4a65-bad8-6bc31207cb8f.png)

![opensea](https://user-images.githubusercontent.com/68864597/151644146-0e6a4483-a5b1-4b79-a33f-35312c731a57.png)

**Comment by PatrickAlphaC** [comment]
Thanks for making this... I can see your contract, I even tried `creating` a new one myself... so odd... I'm not sure.

It looks like you did everything right. I can see your cat and everything...

Sorry, I'd have to debug with you for quite some time to figure it out... It could be an opensea problem... but yeah, it looks like everything is correct... 

<img width="742" alt="Screen Shot 2022-02-01 at 3 14 47 PM" src="https://user-images.githubusercontent.com/54278053/152044140-42d6935b-91d5-4a8d-bfa9-25f439b20bab.png">

Closing for now... but happy to follow up.

**Comment by justinhchae** [comment]
Thanks for the response! I have actually been able to track down the issue. 

The code you provide does exactly what you designed it to do - the coin is indeed minted. 

The part I was missing was to get the right information that represents the actual minted token which can be accessed in Etherscan in the "Tokens Transferred" section. See screen capture below:

<img width="795" alt="get_token" src="https://user-images.githubusercontent.com/68864597/152048873-2455cb3c-ce8b-481e-9421-ef175a96eabf.png">

I
I was able to learn this by following another tutorial from Etherium.org [https://ethereum.org/en/developers/tutorials/how-to-view-nft-in-metamask/](https://ethereum.org/en/developers/tutorials/how-to-view-nft-in-metamask/). The part about transfering a token to wallet after minting. 

Anyway, big thanks again for putting this out there.

**Comment by PatrickAlphaC** [comment]
No problem... You still should be able to see it on opensea after deploying the contract... can you see it in opensea now? What did you do to resolve this?

**Comment by justinhchae** [comment]
Yea, thanks for asking about this becuase the OpenSea part is still unresolved. This is what is interesting / I'm not sure about. 

If I look up the contract address [0xdd73138e76a04f5392c776147f0725a1b7e5867d] I think I see the token that I minted originally and a second token (one that you made more recently using the same contract?) on OpenSea. I also think I see the transaction where I transferred it to my Metamask wallet. 

However, in OpenSea, I only see tokens that are probably test runs based on your tutorial (circles and random lines). But the more complex SVGs like my cat test show up with only the icon of a picture (next screenshot).  I am wondering whether this is because it's one or a combination of a few things (too big/complex image, not supported on testnet, ...)

<img width="1275" alt="testnet-os" src="https://user-images.githubusercontent.com/68864597/152423203-ef608554-d07b-456b-b7d0-87cffaacffee.png">
